### PR TITLE
Fix bug in signature parsing for EIP712TypedData

### LINKED ISSUE #8812: Investigate "invalid code" error (valora/AS)
Receiving an "invalid code" error sporadically for verification flows, that seems to be caused by an authentication error from the logs.

A code that seems valid (sent by attestation service) does not work. Resending the code does not fix this as the same code is resent, so this ends up blocking the verification flow. (It also seems to block it indefinitely -- later on it just says "you're all set" and doesn't allow you to restart the verification flow at all).

### PR DESCRIPTION
**Pull request by eelanagaraj**
## Description

When parsing EIP712 signed typed data in `SignatureUtils.recoverEIP712TypedDataSigner`, we break and return a value as soon as a signature can be parsed. The issue is that 1) we do not serialize signatures consistently (examples from the monorepo: `SignatureUtils.serializeSignature` serializes as VRS, but `LocalSigner.signTypedData` returns an RSV-serialized sig) 2) some small percentage of signatures can be parsed as both RSV & VRS signatures, in which case the order of the parsing matters as long as we do not check against the expected signer.

Given that the usage in `offchain-data-wrapper` is relatively new (and breaking changes shouldn't be too disruptive), the proposed fix splits up `recoverEIP712TypedDataSigner` into two (VRS & RSV-specific, so the caller can choose to explicitly call these if it's necessary to "guess" an address), while modifying the `verify` function to be in line with how the existing message parsing `verifySignature` function works -- i.e. boolean that suppresses errors but works regardless of RSV/VRS serialization.

This is a breaking change in that `recoverEIP712TypedDataSigner` has been deprecated in favor of two functions.

### Other changes
- wraps `SignatureUtils.verifyEIP712TypedDataSigner` in a try/catch to ensure that it always returns a boolean (and no longer propagates the error from the recovery step). This is in line with the other signature parsing/verification functions.
- adds unit tests for `recover & verify`: these are pretty dumb and based on hardcoded examples in order to not add the CK/LocalWallet dependencies to the utils package -- open to thoughts on this though, if you think it's worthwhile to randomize PKs but incorporate these dependencies)

### Testing
- [x] unit tests for `recover & verify` against a hard-coded edge case that was failing
- [x] existing unit tests for `offchain-data-wrapper` pass
- [x] programmatically tested that previous issue with certain AS codes causing this is resolved for large ranges of codes
- [x] TODO deploy to alfajores AS + make sure no changes to existing flow (also tried with contrived code examples that would have thrown the `Invalid signature` error, but now throw "Invalid code" as we want)

Closes #8812

_Edit: further notes for future documentation:_
### Notes on implementation choices in `offchain-data-wrapper` and future TODOs here:

Initially @nambrot and I thought it would make sense to double down on the usage of just VRS-serialization. This PR diverges from that initial thought due to the following:
- in `PublicSimpleAccessor.sign` we are actually using the `wallet.signTypedData` method which returns an RSV-signature (we can switch this to using the `CK.signTypedData` + `SignatureUtils.serializeSignature`, but this increases the scope of this PR)
- we could just use the RSV recover function, but this moves away from compatibility with serializeSignature (VRS)
- (even within the wrapper code, we use both `wallet.signTypedData` and `serializeSignature` (in the tests), so standardizing should take a more rigorous comb through of everything here IMO)

The current implementation is slightly less efficient than before, but should not suffer from the same edge case that this PR fixes (except in the case that the `guessedSignature` must still be used, which seems unavoidable). Notes on this:
- we repeat signature recoveries here (i.e. use the `verifyEIP712TypedDataSigner`) which is less efficient (on the order of a few milliseconds in total)
- if we don’t have a single standard signature, then moving the signature-guessing logic up will either 1) reintroduce the parsing edge case or 2) make it less readable & duplicates logic IMO (try to parse both, check equality against up to 2 parsed sigs)

I ultimately opted for minimizing the amount of signature standardization across the module for now and eliminating the possible edge case, and would be slightly in favor of splitting out signature standardization across the offchain-data-wrapper into a separate ticket for the future.

TODOs
- [x] [immediate] create a ticket for the below and link here -> created issue #8862 (assigned to Applications initially)

### CODE REVIEW

File: packages/sdk/utils/src/signatureUtils.ts [hunk 1]
```diff
@@ -134,37 +134,70 @@ export function parseSignatureWithoutPrefix(
...
 
-export function recoverEIP712TypedDataSigner(
+function recoverEIP712TypedDataSigner(
   typedData: EIP712TypedData,
-  signature: string
+  signature: string,
+  parseFunction: (signature: string) => Signature
 ): string {
   const dataBuff = generateTypedDataHash(typedData)
+  const { r, s, v } = parseFunction(signature.slice(2))
```
Comment (codyborn) [inline packages/sdk/utils/src/signatureUtils.ts]: nit: it would be nice to have a util method that we can use similar to `strip0x` that will only remove the `0x` if it's present.

Comment (codyborn) [reply packages/sdk/utils/src/signatureUtils.ts]: Just found there is one in sdk/base -`trimLeading0x`

**Review by codyborn — APPROVED** [review]
Very thorough per usual! 🚢
